### ISSUE TITLE
Number of mediapipe marker names should match length of mediapipe tracked points in model info

### ISSUE THREAD

**Issue by aaroncherian** [freemocap/freemocap#578]
This is part skellytracker (because that's where the model info lives) and part freemocap (because that's where the skeleton lives). Something I noticed is that in the mediapipe `ModelInfo`:

```
class MediapipeModelInfo(ModelInfo):
    body_landmark_names = [
        landmark.name.lower() for landmark in mp_holistic.PoseLandmark
    ]
    hand_landmark_names = [
        landmark.name.lower() for landmark in mp_holistic.HandLandmark
    ]
    face_landmark_names = [
        "right_eye",
        "left_eye",
        "nose_tip",
        "mouth_center",
        "right_ear_tragion",
        "left_ear_tragion",
    ]
    landmark_names = body_landmark_names + hand_landmark_names + face_landmark_names
    num_tracked_points_body = len(body_landmark_names)
    num_tracked_points_face = FACEMESH_NUM_LANDMARKS_WITH_IRISES
    num_tracked_points_left_hand = len(hand_landmark_names)
    num_tracked_points_right_hand = len(hand_landmark_names)
    num_tracked_points = (
        len(body_landmark_names)
        + 2 * len(hand_landmark_names)
        + num_tracked_points_face
    )
    ```
    
    The `num_tracked_points_face` doesn't match the `face_landmark_names` in length, which means that `len(landmark_names)` wouldn't be equal to `num_tracked_points`. The reason this is important is because in the `Skeleton` class there's currently this check in `integrate_3d_data`:
    
    ```
        def integrate_freemocap_3d_data(self, freemocap_3d_data:np.ndarray):
          num_markers_in_data = freemocap_3d_data.shape[1]
          original_marker_names_list = self.markers.original_marker_names.markers
          num_markers_in_model = len(original_marker_names_list)
          
          if num_markers_in_data != num_markers_in_model:
              raise ValueError(
                  f"The number of markers in the 3D data ({num_markers_in_data}) does not match "
                  f"the number of markers in the model ({num_markers_in_model})."
              )
      
          for i, marker_name in enumerate(original_marker_names_list):
              self.marker_data[marker_name] = freemocap_3d_data[:, i, :]
     ```
     
     Where the skeleton is matching the number of markers in the data given to it with the number of markers in the model, (which is taken from this line  `num_markers_in_model = len(original_marker_names_list)`)
     
  Part of the reason this is happening is because I wrote this initially to just work with the mediapipe body stuff, where we had a strict number of landmarks that we knew the name of, not for a face mesh like situation. There are a couple of solutions I have in mind:
  
 1) We fill the `face_landmark_names` with dummy names after the named ones, until it reaches the length of the mesh (face_001, face_002, that sort of thing). This way, the landmark names length is equal to num_tracked_points.
 2) We alter the check in `integrate_3d_skeleton` to account for or work around this
 3) We talk about the previous idea we had of using separate trackers for the body/hands/face, which is the more involved answer, and would probably also end up being some combination of 1 and 2.
 
 This is also important because OpenPose has a similar body/hands/face situation to MediaPipe, so the best bet would be to figure out a general forward-facing solution.

**Comment by philipqueen** [comment]
Interesting. I think this issue would be bypassed by using `num_tracked_points` from the model info instead of `len(original_marker_names_list)`. 

I *think* from looking at the integrate function this should be fine. We don't want every tracked point to be in the skeleton - we only want the points with markers. Is that correct?

**Comment by philipqueen** [comment]
Pushed a fix for this in #536
